packages/mobile/src/screens/edit-track-screen/fields/VisibilityField/VisibilityScreen.tsx

```diff
@@ -0,0 +1,162 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { visibilityMessages } from '@audius/common/messages'
+import { getLocalTimezone } from '@audius/common/utils'
+import dayjs from 'dayjs'
+import { useFormikContext } from 'formik'
+import { formatCalendarTime } from 'utils/dateUtils'
+
+import {
+  Flex,
+  Hint,
+  IconCalendarMonth,
+  IconVisibilityHidden,
+  IconVisibilityPublic
+} from '@audius/harmony-native'
+import { DateTimeInput } from 'app/components/core'
+import { useNavigation } from 'app/hooks/useNavigation'
+import { FormScreen } from 'app/screens/form-screen'
+
+import { ExpandableRadio } from '../../components/ExpandableRadio'
+import { ExpandableRadioGroup } from '../../components/ExpandableRadioGroup'
+import type { FormValues } from '../../types'
+
+type VisibilityType = 'scheduled' | 'public' | 'hidden'
+
+const messages = {
+  ...visibilityMessages,
+  scheduled: (date: string) => `Scheduled for ${formatCalendarTime(date)}`
+}
+
+export const VisibilityScreen = () => {
+  const { values, initialValues, setValues } = useFormikContext<FormValues>()
+  const { is_unlisted, is_scheduled_release, release_date, isUpload } = values
+  const isAlreadyPublic = !isUpload && !initialValues.is_unlisted
+
+  const initialVisibilityType =
```

Comment:
should this be based on `initialValues`? or is that the original form state and we want the 'initial' state as you open this menu this time?

Reply:
very good eyes! initial is a bit overloaded here... another name is eluding me, maybe initialVisibilityTypeState?